query/src/test/java/org/infinispan/query/distributed/MassIndexingTest.java

```diff
@@ -59,29 +61,33 @@ public void testOverlappingMassIndexers() {
 
       IntStream.range(0, 10).forEach(i -> cache.put(i, new Car("whatever", "whatever", 0)));
 
-      CompletableFuture<Void> first = massIndexer.startAsync();
-      eventually(massIndexer::isRunning);
 
-      CompletableFuture<Void> second = massIndexer.startAsync();
+      CompletionStage<Void> first = massIndexer.startAsync();
+      eventually(() -> {
+         log.debug("Checking if massIndexer is running");
+         return massIndexer.isRunning();
+      }, 10_000, 10, TimeUnit.MILLISECONDS);
```

Comment:
Do you still need the debug log? Maybe it should be trace?

Also poll interval is awfully low, although I'm removing the poll interval parameter altogether in #8344

Reply:
It is low because the test finishes between even 50ms poll timeout and then fails. Like I mentioned in chat, this test is very very flaky. I would love if @gustavonalle could rewrite this with his other changes :)

Reply:
It started failing from me since this PR went in :)
I will rewrite this test in one of my PRs

Reply:
The reason I say it is flaky is because it uses eventually in a way that isn't designed. It wants to see the lock being held, but there is a chance that it can acquire the lock and complete during that time frame.

The only reason it didn't fail before is because it would block waiting for the lock and then it wouldn't finish before it checked.

Eventually is designed (or at least my assumption has been) that it wants the value to be in a stable state eventually (so you can query later at any point).

Reply:
The test used to be stable before, as it had about 1000 entries and was never too fast, but @danberindei complained the test was taken too long with TRACE on, so the number of entries was reduced. I agree this should be reviewed so that it works with any number of entries

Reply:
The best way would be to use some sort of synchronization tool to prevent mass indexer from completing and then wait for that in the main thread - then submit the new task which would always not be able to acquire the lock and then finally release mass indexer to complete in the main thread.